core/accessors.go

```diff
@@ -552,3 +552,71 @@ func WriteRunningEventFilter(w db.KeyValueWriter, filter *RunningEventFilter) er
 
 	return w.Put(db.RunningEventFilter.Key(), enc)
 }
+
+// GetDeprecatedCasmHash retrieves the deprecated casm hash for
+// a given class hash and block number.
+func GetDeprecatedCasmHash(
+	r db.KeyValueReader,
+	blockNum uint64,
+	classHash *felt.Felt,
+) (felt.Felt, error) {
+	var compiledClassHash felt.Felt
+	err := r.Get(
+		db.SierraClassHashToDeprecatedCasmHashKey(blockNum, classHash),
+		func(data []byte) error {
+			compiledClassHash.Unmarshal(data)
+			return nil
+		})
+	if err != nil {
+		return felt.Felt{}, err
+	}
+
+	return compiledClassHash, nil
+}
+
+func WriteDeprecatedCasmHash(
+	w db.KeyValueWriter,
+	blockNum uint64,
+	classHash,
+	compiledClassHash *felt.Felt,
+) error {
+	return w.Put(
+		db.SierraClassHashToDeprecatedCasmHashKey(blockNum, classHash),
+		compiledClassHash.Marshal(),
+	)
+}
+
+func DeleteDeprecatedCasmHash(
+	w db.KeyValueWriter,
+	blockNum uint64,
+	classHash *felt.Felt,
+) error {
+	return w.Delete(db.SierraClassHashToDeprecatedCasmHashKey(blockNum, classHash))
+}
+
+// DeleteDeprecatedCasmHashUpToBlockNum deletes all deprecated casm hash mappings
+// where blockNum <= given blockNum. Useful for pruning old data after L1 finality.
+// This should be called when updating the L1 head to clean up data that is no longer needed
+// for reorg handling.
+func DeleteDeprecatedCasmHashUpToBlockNum(w db.KeyValueRangeDeleter, blockNum uint64) error {
+	bucketPrefix := db.SierraClassHashToDeprecatedCasmHash.Key()
+
+	start := bucketPrefix
+
+	endBlockNum := blockNum + 1
+	end := db.SierraClassHashToDeprecatedCasmHash.Key(MarshalBlockNumber(endBlockNum))
+
+	return w.DeleteRange(start, end)
+}
+
+// DeleteDeprecatedCasmHashFromBlockNum deletes all deprecated casm hash mappings
+// where blockNum >= given fromBlock. Useful for pruning data from a specific block onwards.
+// This should be called when you want to remove all data from a certain block number to the end.
+func DeleteDeprecatedCasmHashFromBlockNum(w db.KeyValueRangeDeleter, fromBlock uint64) error {
+	start := db.SierraClassHashToDeprecatedCasmHash.Key(MarshalBlockNumber(fromBlock))
+
+	maxBlockNum := ^uint64(0) // max uint64
```

Comment:
Just in case you didn't know, there is a `math.MaxUint64` constant, but I am happy with this cool initialization as well